File: back-kotlin/src/main/kotlin/me/nicolas/stravastats/domain/business/SegmentClimbProgression.kt
package me.nicolas.stravastats.domain.business

data class SegmentClimbProgression(
    val metric: String,
    val targetTypeFilter: String,
    val weatherContextAvailable: Boolean,
    val targets: List<SegmentClimbTargetSummary>,
    val selectedTargetId: Long?,
    val selectedTargetType: String?,
    val attempts: List<SegmentClimbAttempt>,
)

data class SegmentClimbTargetSummary(
    val targetId: Long,
    val targetName: String,
    val targetType: String,
    val climbCategory: Int,
    val distance: Double,
    val averageGrade: Double,
    val attemptsCount: Int,
    val bestValue: String,
    val latestValue: String,
    val consistency: String,
    val averagePacing: String,
    val closeToPrCount: Int,
    val recentTrend: String,
)

data class SegmentClimbAttempt(
    val targetId: Long,
    val targetName: String,
    val targetType: String,
    val activityDate: String,
    val elapsedTimeSeconds: Int,
    val speedKph: Double,
    val distance: Double,
    val averageGrade: Double,
    val elevationGain: Double,
    val prRank: Int?,
    val setsNewPr: Boolean,
    val closeToPr: Boolean,
    val deltaToPr: String,
    val weatherSummary: String?,
    val activity: ActivityShort,
)

File: back-go/domain/business/segmentclimbprogression.go
package business

type SegmentClimbProgression struct {
	Metric                  string
	TargetTypeFilter        string
	WeatherContextAvailable bool
	Targets                 []SegmentClimbTargetSummary
	SelectedTargetId        *int64
	SelectedTargetType      *string
	Attempts                []SegmentClimbAttempt
}

type SegmentClimbTargetSummary struct {
	TargetId       int64
	TargetName     string
	TargetType     string
	ClimbCategory  int
	Distance       float64
	AverageGrade   float64
	AttemptsCount  int
	BestValue      string
	LatestValue    string
	Consistency    string
	AveragePacing  string
	CloseToPrCount int
	RecentTrend    string
}

type SegmentClimbAttempt struct {
	TargetId           int64
	TargetName         string
	TargetType         string
	ActivityDate       string
	ElapsedTimeSeconds int
	SpeedKph           float64
	Distance           float64
	AverageGrade       float64
	ElevationGain      float64
	PrRank             *int
	SetsNewPr          bool
	CloseToPr          bool
	DeltaToPr          string
	WeatherSummary     *string
	Activity           ActivityShort
}
